nvtabular/dask/io.py

```diff
@@ -178,42 +192,56 @@ def __init__(self, path, engine, part_size=None, part_mem_fraction=0.125, **kwar
                 )
             part_size = int(cuda.current_context().get_memory_info()[1] * part_mem_fraction)
 
+        # Engine-agnostic path handling
+        if hasattr(path, "name"):
+            path = stringify_path(path)
+        storage_options = storage_options or {}
+        fs, fs_token, paths = get_fs_token_paths(path, mode="rb", storage_options=storage_options)
+        paths = sorted(paths, key=natural_sort_key)
+
+        # If engine is not provided, try to infer from end of paths[0]
+        if engine is None:
+            engine = paths[0].split(".")[-1]
+
         if isinstance(engine, str):
             if engine == "parquet":
-                self.engine = ParquetDatasetEngine(path, part_size, **kwargs)
+                self.engine = ParquetDatasetEngine(paths, part_size, fs, fs_token, **kwargs)
+            elif engine == "csv":
+                self.engine = CSVDatasetEngine(paths, part_size, fs, fs_token, **kwargs)
             else:
-                raise ValueError("Only parquet supported for now")
+                raise ValueError("Only parquet and csv supported (for now).")
         else:
-            self.engine = engine(path, part_size, **kwargs)
-
-    def meta_empty(self, columns=None):
-        return self.engine.meta_empty(columns=columns)
+            self.engine = engine(paths, part_size, fs, fs_token, **kwargs)
 
     def to_ddf(self, columns=None):
         return self.engine.to_ddf(columns=columns)
 
 
 class DatasetEngine:
-    """ DaskDataset Class
-        Converts dataset `pieces` to a dask_cudf DataFrame
+    """ DatasetEngine Class
+
+        Base class for Dask-powered IO engines. Engines must provide
+        a ``to_ddf`` method.
     """
 
-    def __init__(self, path, part_size):
-        if hasattr(path, "name"):
-            path = stringify_path(path)
-        fs, _, paths = get_fs_token_paths(path, mode="rb")
-        self.fs = fs
-        self.paths = sorted(paths, key=natural_sort_key)
+    def __init__(self, paths, part_size, fs, fs_token):
+        self.paths = paths
         self.part_size = part_size
-
-    def meta_empty(self, columns=None):
-        raise NotImplementedError(""" Return an empty cudf.DataFrame with the correct schema """)
+        self.fs = fs
+        self.fs_token = fs_token
 
     def to_ddf(self, columns=None):
         raise NotImplementedError(""" Return a dask_cudf.DataFrame """)
 
 
 class ParquetDatasetEngine(DatasetEngine):
+    """ ParquetDatasetEngine
+
+        Dask-based version of cudf.read_parquet.
+        TODO: Improve dask_cudf.read_parquet performance so that
+        this class can be slimmed down.
```

Comment:
We should keep TODO's out of docstrings (which get pulled out for our user facing documentation).

I'd leave this as a comment instead, and link to an issue on dask_cudf about improving performance.

Reply:
Good catch - My mistake